tests/test_gradient_detaching.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+import torch
+import xarray as xr
+
+from ocean_emulators.config import SamudraConfig, UNetBackboneConfig
+from ocean_emulators.constants import TensorMap
+from ocean_emulators.datasets import TrainData
+from ocean_emulators.utils.data import DataSource, Normalize
+from ocean_emulators.utils.multiton import MultitonScope
+
+
+@pytest.fixture
+def samudra_setup():
+    with MultitonScope():
+        # Set up minimal data structures needed by Samudra
+        h, w = 8, 8
+        coords = {
+            "lev": [0],
+            "lat": (["y"], torch.linspace(-90, 90, h).numpy()),
+            "lon": (["x"], torch.linspace(-180, 180, w).numpy()),
+        }
+        data = xr.Dataset(
+            {
+                "thetao": (["lev", "y", "x"], torch.randn(1, h, w).numpy()),
+                "hfds": (["y", "x"], torch.randn(h, w).numpy()),
+            },
+            coords=coords,
+        )
+        ones = xr.Dataset(
+            {
+                "thetao": (["lev", "y", "x"], torch.ones(1, h, w).numpy()),
+                "hfds": (["y", "x"], torch.ones(h, w).numpy()),
+            },
+            coords=coords,
+        )
+        src = DataSource(name="dummy", data=data, means=data, stds=ones)
+
+        # Initialize TensorMap and Normalize
+        TensorMap.init_instance("thetao_1", "hfds")
+        Normalize.init_instance(
+            src,
+            TensorMap.get_instance().prognostic_var_names,
+            TensorMap.get_instance().boundary_var_names,
+            torch.ones(h, w),
+            torch.ones(h, w),
+        )
+
+        # Create Samudra model configuration
+        config = SamudraConfig(
+            unet=UNetBackboneConfig(
+                ch_width=[4, 8],
+                dilation=[1, 1],
+                n_layers=[1, 1],
+            ),
+            pos_channels=0,
+        )
+
+        def create_samudra_model(gradient_detach_interval=0):
+            # Set the gradient_detach_interval on the config object
+            config.gradient_detach_interval = gradient_detach_interval
+            return config.build(
```

Comment:
🐑 instead of changing the gradient detach attr by side effect, I think it's safe to simply move the config instantiation code inside of this function. 
```suggestion

        def create_samudra_model(gradient_detach_interval=0):
            config = SamudraConfig(
                unet=UNetBackboneConfig(
                    ch_width=[4, 8],
                    dilation=[1, 1],
                    n_layers=[1, 1],
                ),
                pos_channels=0,
                gradient_detach_interval=gradient_detach_interval,
            )
            return config.build(
```

Reply:
If you want to be really suave, you could do this, too:

```suggestion

        def create_samudra_model(gradient_detach_interval=0):
            return SamudraConfig(
                unet=UNetBackboneConfig(
                    ch_width=[4, 8],
                    dilation=[1, 1],
                    n_layers=[1, 1],
                ),
                pos_channels=0,
                gradient_detach_interval=gradient_detach_interval,
            ).build(
```